### PR TITLE
Fedexd

### PR DESCRIPTION
**Pull request by FedeGaleano**
Decouple Inventory from Scene
Scene as observator of Subscene Input
Stop passing Input in Scene family constructors
Abstract class for Gameplay Scenes (will be renamed after merging) containing common gameplay data
GameState for commumication between Gameplay-Scenes 
Reset Camera before re-entering the scene
DialogBox UI class
Re-modeled ShipScene and added hatch + crafter as interaction elements

### CODE REVIEW

File: TGC.Group/Model/Scenes/InventoryScene.cs [hunk 1]
```diff
@@ -34,20 +34,23 @@ private TGCVector2 GetScaleForSpriteByPixels(CustomSprite sprite, int xPixels, i
...
             {
                 SetState(nextStateID);
                 nextStateID = StateID.NULL;
             }
             updateLogic(elapsedTime);
+
         }
         public override void Render()
         {
+            if (stateID == StateID.CLOSED) return;
```
Comment (ankomas) [inline TGC.Group/Model/Scenes/InventoryScene.cs:47]: turbinol

File: TGC.Group/Model/Scenes/ShipScene.cs [hunk 2]
```diff
@@ -123,5 +204,48 @@ public ShipScene OnPause(Callback onPauseCallback)
...
+        //}
+        //public void CloseCrafter()
+        //{
+        //    subScene = Scene.Empty;
+        //    TurnExploreCommandsOn();
+        //    ((Camera)Camera).ConsiderInput();
+        //}
+    }
+
+    class Thing
```
Comment (ankomas) [inline TGC.Group/Model/Scenes/ShipScene.cs:226]: thing xd

Comment (JuanAmarillo) [reply TGC.Group/Model/Scenes/ShipScene.cs:226]: jjajajajj

File: TGC.Group/Model/UI/DialogBox.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+using System.Threading.Tasks;
+using TGC.Core.Mathematica;
+using TGC.Core.Text;
+using TGC.Group.Model.Resources.Sprites;
+using TGC.Group.Model.Utils;
+using TGC.Group.TGCUtils;
+
+namespace TGC.Group.Model.UI
+{
+    class DialogBox
```
Comment (ankomas) [inline TGC.Group/Model/UI/DialogBox.cs:15]: llevo esperando esta clase desde que empezamos )?

**Review by ankomas — APPROVED** [review]

**Pull request merged**